src/main/java/com/microsoft/azure/vmagent/remote/AzureVMAgentSSHLauncher.java

```diff
@@ -316,7 +343,7 @@ private int executeRemoteCommand(final Session jschSession, final String command
         try {
             // If root, modify the command to set up sudo -S
             String finalCommand = null;
-            if (executeAsRoot) {
+            if (isUnix && executeAsRoot) {
```

Comment:
We have several checks on `isUnix` and a few other states. I wonder if we can create an abstract template base class with abstract methods that provide platform dependent properties / behaviors, and then create two concrete subclass for Unix and Windows respectively.